Cpp_hands_on_programming/documentation/error_handling.md

```diff
@@ -0,0 +1,44 @@
+# Possible errors
+- [Possible errors](#possible-errors)
+  - [Summary](#summary)
+  - [Option combination](#option-combination)
+    - [Invalid option](#invalid-option)
+    - [Invalid argument](#invalid-argument)
+      - [IPC methods](#ipc-methods)
+      - [File](#file)
+  - [Timeout.](#timeout)
+    - [Only one process launched](#only-one-process-launched)
+    - [Two different IPC methods launched](#two-different-ipc-methods-launched)
+    - [One process terminated during run time](#one-process-terminated-during-run-time)
+## Summary
+This program is a file transfering program. There are two processes required to be launched in parallel so that the transfering is possible, i.e sender and receiver.   
+Program throws an error and exit the program if one of the below scenarios happen .
+## Option combination
+### Invalid option
+- No `--file/-f` option is provided
+- No or more than one method option is provided, e.g `--pipe --queue` 
+- Duplicate option, e.g `--shm --shm` 
+### Invalid argument
+#### IPC methods
+The argument is the name of IPC method object being used, e.g `--shm /shmName`
+- Argument is not provided for argument required option, e.g pipe, queue, shm
+- Invalid name according to IPC method, e.g [message queue naming rule](https://man7.org/linux/man-pages/man7/mq_overview.7.html)
+
```

Comment:
If the IPC name is too long we have an error "file name is too long" but... it's not the filename.

Reply:
Yes because it is the default description of ENAMETOOLONG, combined with the prefix set in the program e,g Ipc_send::mq_open:  file name is too long.  I think the information is solid enough for user to find out the cause